rename_files.py

```diff
@@ -33,14 +33,22 @@ def main():
 
 
 def get_input():
+    print(" You entered", len(sys.argv)-1, "arguments at the command line.")
+
+    if len(sys.argv) != 4:
+        # IndexError: list index out of range
+        print(" Error: Wrong number of arguments. Enter 3 arguments: 1. "
+              "string to replace 2. replacement string 3. path for files ")
+        sys.exit(1)
+
```

Comment:
This works but it has the drawback that a decision to exit the program is buried in a function, and most people advise putting that kind of thing in the main block.

To do that, you'd change your check to raise an exception with a nicer error message:

```
if len(sys.argv) != 4:
    raise Exception("Error: Wrong number of arguments ...")
```

If you leave it like that, when the user gives the wrong number of arguments, there will still be a stack trace, but the message will explain the problem in terms that make more sense to the user.